### PR TITLE
Add review comments to docs, fix typos, improve style

### PR DESCRIPTION
**Pull request by philipstanislaus**

### CODE REVIEW

**Review by vgeddes — APPROVED** [review]
LGTM!

**Comment by philipstanislaus** [comment]
@musnit any thoughts on this?

File: docs/markdown/concepts/polkadot-light-client/interactive-protocol.md [hunk 1]
```diff
@@ -6,23 +6,28 @@ permalink: /concepts/polkadot-light-client-verifier/interactive-protocol
...
 
-To convince the chain that at least 1/3 of validators voted for something, we will need to sample validators who are claimed to vote for this at random. We will use a block hash for this purpose, which means we need to deal with the issue that this is influenceable. Since we are working with the assumption that at least 2/3 of validators are honest and up to almost 1/3 might be dishonest, we can only expect 2/3 of validators to vote for something. If only exactly $(n-1)/3 + 1$ validators vote for something, then we cannot prove that at least $(n-1)/3 + 1$ did by random sampling. Instead we will need a vote where all honest validators vote for something and it is enough to show that a single honest validator voted for it, for which it suffices to argue that over 1/2 of these 2/3 claimed votes are for it. This means that the validators must sign something after they know it is already final, which is an extra signature/vote beyond just that for Byzantine agreement.
+<!-- TODO: the above is quite confusing. How can we better describe that? -->
```
Comment (musnit) [inline docs/markdown/concepts/polkadot-light-client/interactive-protocol.md:16]: yeh this page is mostly copied from web3's document and so is pretty technical/difficult to understand. we can rewrite it in future

File: docs/markdown/concepts/polkadot-light-client/interactive-protocol.md [hunk 2]
```diff
@@ -6,23 +6,28 @@ permalink: /concepts/polkadot-light-client-verifier/interactive-protocol
...
 ## The interactive protocol
 
 A prover wants to convince a light client that at least $1/3$ of validators signed a statement, which they claim that a specific set of at least $2/3$ of validators do. We assume that the light client already has a Merkle root $r_{val}$ of validator public keys.
 
 1. The prover sends to the light client the statement $S$, a bitfield $b$ of validators claimed to sign it (which claims that more than $2/3$ of validators signed $S$), one signatures $sig$ on $S$ from an arbitrary validator together with Merkle proofs from $r_{val}$ of their public key.
 
-2. The light client verifies the backing signature and its proof and asks for $k_{approval}$ validators at random among those that $b$ claims signed $S$
+2. The light client checks that the bitfield claims at least 2/3 of validators signed. It then verifies the backing signature and its proof and asks for $k_{approval}$ validators at random among those that $b$ claims signed $S$
+
+<!-- TODO: that will happen off-chain, right? The sampling should come from a later block hash, which the light client does not yet know -->
```
Comment (musnit) [inline docs/markdown/concepts/polkadot-light-client/interactive-protocol.md:28]: yeh, flow could be rewritten

File: docs/markdown/concepts/polkadot-light-client/interactive-protocol.md [hunk 3]
```diff
@@ -38,24 +43,32 @@ In order to have an adversary with access to sufficient hashpower on Ethereum st
 
 Now, if we e.g. assume that over $2/3$ of the hashpower of Ethereum is owned by honest miners and that over $1/2$ of the claimed signers are honest validators who didn't sign $S$, then we can analyse the maximum probability that the 100th block after the first transaction was included has a blockhash that results in the test succeeding against any strategy by the adversarial miners. This will involve building a Markov chain model of the worst case and proving that the bad guys can't do better. A back of the envelope calculation gave me that this would be something like $7p/2$ chance of success vs $p$ for a random number.
 
+<!-- TODO: is that your analysis @musnit or is it taken from another document? -->
```
Comment (musnit) [inline docs/markdown/concepts/polkadot-light-client/interactive-protocol.md:46]: this is all copied from web3's research doc

File: docs/markdown/concepts/polkadot-light-client/interactive-protocol.md [hunk 4]
```diff
@@ -38,24 +43,32 @@ In order to have an adversary with access to sufficient hashpower on Ethereum st
...
+
 Now we'd rather argue about rational miners than honest ones. In this case, producing a block with a hash that fails the test, which happens with probability $1-p$, would gain the miner some block reward $R$ if they released it. It would cost them in expectation $(1-p)/p$ block rewards. With $p=2^{-k_{approval}}$, this is $(2^{k_{approval}}-1)R$. With $R=5$ ETH and $k_{approval} = 25$, this would be 167,772,155 ETH which is more than the 112,421,804 ETH currently in existence. Something like this would be secure enough for rational miners to be honest even if there was only one mining pool for Ethereum.
 
+<!--  TODO: the above section should go into a separate section on economic analysis, and not be under the implementation heading -->
```
Comment (musnit) [inline docs/markdown/concepts/polkadot-light-client/interactive-protocol.md:50]: yeh probably, this whole page needs a rewrite at some point.

File: docs/markdown/concepts/polkadot-light-client/mmr-light-client.md [hunk 5]
```diff
@@ -6,12 +6,15 @@ permalink: /concepts/polkadot-light-client-verifier/mmr-light-client
...
-After the interactive protocol runs, we have new [BEEFY](https://github.com/paritytech/grandpa-bridge-gadget) MMR commitments. These are merkle mountain ranges that contain data for updates to the Polkadot Validator set and data for new parachain headers.
+After the interactive protocol runs, we have new [BEEFY](https://github.com/paritytech/grandpa-bridge-gadget) MMR commitments. These are the root hashes of merkle mountain ranges that contain data for updates to the Polkadot Validator set and data for new relay chain headers.
 
 We use merkle proofs to verify the contents of each BEEFY MMR, extracting the above two kinds of data.
 
 New validator set updates are applied to our interactive protocol client so that it knows which validator signatures will be valid for the next block.
 
 The new Parachain header for our Snowbridge chain is verified and then provided to our [Parachain Light Client](/concepts/polkadot-light-client-verifier/parachain-light-client) to be used for verifying and processing new bridge messages.
+
+<!-- TODO: why do we need to store those headers? It should be cheaper to just store the MMR root and then verify a proof against that. -->
```
Comment (musnit) [inline docs/markdown/concepts/polkadot-light-client/mmr-light-client.md:20]: hmm yeh, i think it depends on how things go in practice, like, if we are doing batch processing of parachain commitments, then we just can just do a single MMR verification to process a full commitment without storing the parachain/relay chain header. if we are doing individual processing of each message, for example in a merkle tree commitment, then we'd need to re-verify the MMR each time if we don't store it?

File: docs/markdown/concepts/polkadot-light-client/parachain-commitments.md [hunk 6]
```diff
@@ -6,8 +6,11 @@ permalink: /concepts/polkadot-light-client-verifier/parachain-commitments
...
 grand_parent: Concepts and Architecture
 ---
+
 # Parachain Commitments
 
 Our parachain has a custom commitment scheme for committing to bridge messages on our parachain and placing those commitments into our parachain header along with a custom light client for our parachain that runs on Ethereum.
 
 More details coming soon...
+
+<!-- TODO: can paste our section here from our Google doc outlining the trade-offs -->
```
Comment (musnit) [inline docs/markdown/concepts/polkadot-light-client/parachain-commitments.md:16]: https://github.com/Snowfork/polkadot-ethereum/issues/197

File: docs/markdown/concepts/polkadot-light-client/parachain-light-client.md [hunk 7]
```diff
@@ -6,10 +6,13 @@ permalink: /concepts/polkadot-light-client-verifier/parachain-light-client
...
 # Snowbridge Parachain Light Client
 
 The parachain light client receives our latest parachain block headers from the MMR Light Client. With them, it can extract our parachain commitments.
 
-We can use these commitments to verify and process every message that has been commited to by the bridge.
+We can use these commitments to verify and process every message that has been committed to by the bridge.
 
 More details coming soon...
+
+<!-- TODO: again, storing them is very expensive. Just go up in the proof to the MMR root is much cheaper.  -->
```
Comment (musnit) [inline docs/markdown/concepts/polkadot-light-client/parachain-light-client.md:18]: see previous comment and reply plz?

**Pull request merged**